tests/rule_based_profiler/parameter_builder/test_parameter_builder.py

```diff
@@ -0,0 +1,486 @@
+    parameters: Dict[str, ParameterContainer] = {
+        domain.id: parameter_container,
+    }
+
+    my_evaluation_dependency_0_parameter_builder.build_parameters(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+        batch_request=None,
+        runtime_configuration=None,
+    )
+    assert my_evaluation_dependency_0_parameter_builder.call_count == 1
+
+    my_evaluation_dependency_1_parameter_builder.build_parameters(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+        batch_request=None,
+        runtime_configuration=None,
+    )
+    assert my_evaluation_dependency_1_parameter_builder.call_count == 1
+
+    dependencies_fully_qualified_parameter_names: List[
+        str
+    ] = get_fully_qualified_parameter_names(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+    )
+
+    my_dependent_parameter_builder.resolve_evaluation_dependencies(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+        fully_qualified_parameter_names=dependencies_fully_qualified_parameter_names,
+        runtime_configuration=None,
+    )
+
+    all_fully_qualified_parameter_names: List[
+        str
+    ] = get_fully_qualified_parameter_names(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+    )
+
+    """
+    These assertions show that both "ParameterBuilder" dependencies computed their values, but dependent
+    "ParameterBuilder" has not yet computed its value.
+    """
+    assert (
+        my_evaluation_dependency_0_parameter_builder.raw_fully_qualified_parameter_name
+        in dependencies_fully_qualified_parameter_names
+        and my_evaluation_dependency_0_parameter_builder.json_serialized_fully_qualified_parameter_name
+        in dependencies_fully_qualified_parameter_names
+    )
+    assert (
+        my_evaluation_dependency_1_parameter_builder.raw_fully_qualified_parameter_name
+        in dependencies_fully_qualified_parameter_names
+        and my_evaluation_dependency_1_parameter_builder.json_serialized_fully_qualified_parameter_name
+        in dependencies_fully_qualified_parameter_names
+    )
+    assert (
+        my_dependent_parameter_builder.raw_fully_qualified_parameter_name
+        not in dependencies_fully_qualified_parameter_names
+        or my_dependent_parameter_builder.json_serialized_fully_qualified_parameter_name
+        not in dependencies_fully_qualified_parameter_names
+    )
+    """
+    No "ParameterBuilder" dependencies need to be executed, because None are specified (hence,
+    "my_dependent_parameter_builder.resolve_evaluation_dependencies()" has no effect), but their values exist, because
+    they were executed individually previously.  Thus, set of computed fully_qualified_parameter_names does not change.
+    """
+    assert sorted(all_fully_qualified_parameter_names) == sorted(
```

Comment:
what action did we take to make this true? Are we really testing `resolve_evaluation_dependencies` here?

Furthermore, if that _is_ what we are testing here, then I'd consider moving it to a separate test. When a unit test fails I want to know exactly where to look.

Reply:
@donaldheppner Indeed here, the fact that `resolve_evaluation_dependencies()` runs inside `build_parameters()` -- please see earlier explanation -- is tested.  I was reluctant in putting it into separate test, because shared memory would need to be set up -- and it looks like we are trying to reduce setup.  Do you have a recommendation?  Thanks.